leak-detect_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```

Comment:
As the current implementation doesn't maintain state info across calls these won't be useful in case of integration test scenarios. So made changes so that the state information can be optionally carried across.